src/domain/RestaurantList.ts

```diff
@@ -0,0 +1,106 @@
+import { LOCAL_STORAGE_KEY } from '../constants';
+
+type Category = '한식' | '중식' | '일식' | '아시안' | '양식' | '기타';
+
+type Distance = 5 | 10 | 15 | 20 | 30;
+
+type CategoryAll = '전체' | Category;
+
+type SortTypeAll = 'name' | 'distance';
+
+export interface Restaurant {
+  category: Category;
+  name: string;
+  distance: Distance;
+  description?: string;
+  link?: string;
+  isFavorite: boolean;
+}
+
+const RestaurantList = {
+  add: (restaurant: Restaurant): void => {
+    const restaurants = RestaurantList.getLocalStorage();
+    const updatedRestaurants = [...restaurants, restaurant];
+
+    RestaurantList.updateRestaurants(updatedRestaurants);
+  },
+
+  delete: (name: string) => {
+    const restaurants = RestaurantList.getList();
+    const findIndex = restaurants.findIndex(
+      (restaurant) => restaurant.name === name
+    );
+
+    if (findIndex === -1) return;
+
+    const updatedRestaurat = restaurants;
+    updatedRestaurat.splice(findIndex, 1);
+
+    RestaurantList.updateRestaurants(updatedRestaurat);
+  },
+
+  filterByCategory: (restaurantList: Restaurant[], category: CategoryAll) => {
+    return restaurantList.filter(
+      (restaurant) => restaurant.category === category
+    );
+  },
+
+  getLocalStorage() {
+    return JSON.parse(window.localStorage.getItem(LOCAL_STORAGE_KEY) || '');
+  },
+
+  getList: (
+    category: CategoryAll = '전체',
+    type: SortTypeAll = 'name'
+  ): Restaurant[] => {
+    const restaurants = RestaurantList.getLocalStorage();
+
+    if (category === '전체') {
+      return RestaurantList.sortByType(restaurants, type);
+    }
+
+    const filteredCategory = RestaurantList.filterByCategory(
+      restaurants,
+      category
+    );
+
+    return RestaurantList.sortByType(filteredCategory, type);
+  },
+
+  sortByType: (restaurantList: Restaurant[], type: SortTypeAll) => {
+    if (type === 'distance') {
+      return [...restaurantList].sort(
+        (aRestaurant, bRestaurant) =>
+          aRestaurant.distance - bRestaurant.distance
+      );
+    }
+
+    return [...restaurantList].sort((aRestaurant, bRestaurant) => {
+      return aRestaurant.name.localeCompare(bRestaurant.name);
+    });
```

Comment:
```suggestion
    const compare: (a: Restaurant, b: Restaurant) => number =
      type === 'distance'
        ? (aRestaurant, bRestaurant) =>
            aRestaurant.distance - bRestaurant.distance
        : (aRestaurant, bRestaurant) => {
            return aRestaurant.name.localeCompare(bRestaurant.name);
          };

    return [...restaurantList].sort(compare);
```

if 와 같은 조건**문** 보다 표현**식**을 사용하는 방향으로 코드를 작성하시는게 좋습니다

Reply:
제가 생각해보지 못한 방식을 제시해주셨네요 ! 👍  너무 좋은 코드 알려주셔서 감사합니다 동동